lkmpg.tex

```diff
@@ -228,35 +228,85 @@ \subsection{The Simplest Module}
 	make -C /lib/modules/$(shell uname -r)/build M=$(PWD) clean
 \end{code}
 
+In \verb|Makefile|, \verb|$(CURDIR)| can set to the absolute pathname of the current working directory(after all \verb|-C| options are processed, if any). See more about \verb|CURDIR| in \href{https://www.gnu.org/software/make/manual/make.html}{GNU make manual}.
+
 And finally just run \verb|make| directly.
+
 \begin{codebash}
 make
 \end{codebash}
-If there is no \verb|PWD := $(CURDIR)| statement in Makefile, note that using \verb|sudo make| might not compile. Because the \verb|sudo| will fork the process. According to the Process Model section in \href{https://man7.org/linux/man-pages/man8/sudo.8.html}{sudo manual}, some of the environment variables won't be inherited. Here is the \verb|PWD| variable which will be passed to the kernel Makefile. For example, here is the simple Makefile:
+
+If there is no \verb|PWD := $(CURDIR)| statement in Makefile, note that using \verb|sudo make| might not compile correctly. 
```

Comment:
Check the grammar for the use of "compile" here.

Reply:
Thanks, I check its grammar by using Grammarly, it tells me there are no grammar problems in this sentence.

Reply:
No, Grammarly failed to check the subject in the above sentence. We usually say, someone "compiles" the code. However, the term "using sudo make" should not be the subject of SVO (subject-verb-object) rule.

Reply:
Thanks for your explanation. How about changing it to the following sentence?

If there is no `PWD := $(CURDIR)` statement in Makefile, then it may not compile correctly when using `sudo make`.

Reply:
Change "when using sudo make" to "with `sudo make`".

Reply:
Thanks for your advice.